blockstore/blockstore.go

```diff
@@ -351,3 +381,21 @@ func (bs *gclocker) PinLock(_ context.Context) Unlocker {
 func (bs *gclocker) GCRequested(_ context.Context) bool {
 	return atomic.LoadInt32(&bs.gcreq) > 0
 }
+
+func doProvideManyHashes(ctx context.Context, r routing.ContentProviding, keys []multihash.Multihash) error {
```

Comment:
nit:  reusable func that is used here and in  `provider/reprovider.go#doProvideMany`?

Reply:
No, provideMany is only used here outside of reprovider. I don't want to import the reprovider system here. I don't think it causes dep-cycles anymore but relying on the ContentProvider interface only is cleaner.